clash-ghc/src-ghc/Clash/GHC/Evaluator.hs

```diff
@@ -339,11 +339,13 @@ instantiate _tcm (TyLambda x e) ty m =
   subst  = extendTvSubst subst0 x ty
   subst0 = mkSubst iss0
   iss0   = mkInScopeSet (freeVarsOf e `unionUniqSet` freeVarsOf ty)
-instantiate tcm pVal@(PrimVal (PrimInfo{primType}) tys []) ty m
+instantiate tcm pVal@(PrimVal (PrimInfo{primType}) tys es) ty m
   | isUndefinedXPrimVal pVal
-  = setTerm (TyApp (Prim NP.undefinedX) (piResultTys tcm primType (tys ++ [ty]))) m
+  = setTerm (TyApp (Prim NP.undefinedX) (piResultTys tcm primType (tys ++ esTys ++ [ty]))) m
   | isUndefinedPrimVal pVal
-  = setTerm (TyApp (Prim NP.undefined) (piResultTys tcm primType (tys ++ [ty]))) m
+  = setTerm (TyApp (Prim NP.undefined) (piResultTys tcm primType (tys ++ esTys ++ [ty]))) m
```

Comment:
Maybe it's just because I haven't really worked on the evaluator, but I find this really difficult to read. The function's documentation doesn't really help me either. Given that it's under-documented as is, I wouldn't be opposed to merging this, but it would be great if you could add a comment anyway.